### PR TITLE
[Testing, Tooling] test: run pkg integration tests

### PR DESCRIPTION
**Pull request by bryanchriswhite**
### Human Summary

- [x] Add step to `run-e2e-test-job-template.yaml`
- [x] Add step to `test_e2e` makefile target
- [x] Update testutil hostnames to use in-tilt hostnames
- [ ] Ensure mocks are generated before running E2E tests in CI

### Summary generated by Reviewpad on 01 Dec 23 13:10 UTC
This pull request includes several changes:

1. The patch updates the Makefile to include e2e tests in the `test_e2e` make target. It adds a new line of code to run e2e tests in the `./e2e/pkg/...` directory.

2. The patch modifies the `localnet.go` file by updating the mock node flag with the in-tilt host. It changes the value of the `FlagNode` flag to `"tcp://poktroll-sequencer:36657"`.

3. The patch updates the `CometLocalWebsocketURL` constant in the `localnet.go` file to use the in-tilt host. It changes the value of the constant to `"ws://poktroll-sequencer:36657/websocket"`.

4. The patch fixes a typo in the Makefile. It changes the path from `./e2e/pkg/...` to `./pkg/...` in the `test_e2e` target.

5. The patch updates the CI workflow file to run e2e tests on the `./pkg/...` directory. It adds a new line of code to run e2e tests in the `./pkg/...` directory.

Overall, these changes aim to include the e2e tests in the `test_e2e` make target and update the necessary flags and URLs to use the in-tilt host.

## Issue

- N/A

I'm recalling that some end-to-end tests currently exist inside of some of the packages in the pkg directory.

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Run all unit tests**: `make go_develop_and_test`
- [ ] **Verify Localnet manually**: See the instructions [here](TODO: add link to instructions)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -140,6 +140,7 @@ test_e2e: ## Run all E2E tests
 	export APPGATE_SERVER=$(APPGATE_SERVER) && \
 	POKTROLLD_HOME=../../$(POKTROLLD_HOME) && \
 	go test -v ./e2e/tests/... -tags=e2e
+	go test -v ./e2e/pkg/... -tags=e2e
```
Comment (bryanchriswhite) [inline Makefile:143]:
```suggestion
	go test -v ./pkg/... -tags=e2e
```

**Comment by bryanchriswhite** [comment]
Pumping the brakes on this. I'm realizing that these tests are actually "integration" tests that just happen to be written to run localnet; however, that need not be the case. Based on discussions with @h5law around testing in #220, it's becoming clear that we should prefer to update these integration tests to use an in-memory test network rather than localnet. Benefits:

1. **Improved test isolation**: On-chain state changes are made on independent in-memory networks
2. **Simplified/increased control over test setup**: Test code can do set up programmatically, directly (i.e. no network I/O).
3. **Adds use cases for and consumers of in-memory test network helpers**: Drives improved test helper design and makes usage more ubiquitous.

**Comment by Olshansk** [comment]
@bryanchriswhite When we do decide to look at this again, I do want us to spend some time investigating the simulator [1]. I want to understand if it does or does not fit our use case before we build anything ourselves.

Again, not something we need to look into in the short-term though.

[1] https://docs.cosmos.network/main/learn/advanced/simulation

**Comment by Olshansk** [comment]
@bryanchriswhite Can we close this out?

**Pull request closed without merging**